ecmd-core/dll/ecmdDllCapi.C

```diff
@@ -5665,24 +5659,23 @@ uint32_t readScandefHash(const ecmdChipTarget & target, const char* i_ringName,
 
     /* Let's see if we have already looked up this info */
     /* find scandef file */
-    rc = dllQueryFileLocation(target, ECMD_FILE_SCANDEF, l_filePairs, l_version);
+    rc = dllQueryFileLocation(target, ECMD_FILE_SCANDEF, l_fileLocs, l_version);
     if (rc)
     {
         dllRegisterErrorMsg(rc, "readScandefHash", "Error occured locating scandef file.\n");
         return rc;
     }
 
-    if (findLatchInCache(l_filePairs, latchHashKey64, ringName, o_latchdata))
+    if (findLatchInCache(l_fileLocs, latchHashKey64, ringName, o_latchdata))
     {
         /* We're done, get out of here */
         return rc;
     }
 
-    std::list<std::pair<std::string, std::string> >::iterator l_filePair = l_filePairs.begin();
-    while (l_filePair != l_filePairs.end())
+    for (auto l_fileLoc : l_fileLocs)
```

Comment:
This is going to cause problems with some of the old 32 bit compilers for architectures we support.